recipes/riscv64-native/run.sh

```
@@ -0,0 +1,85 @@
+fi
+
+# Ensure remote directories exist
+ssh ${SSH_OPTS} "${REMOTE_USER}@${REMOTE_HOST}" "mkdir -p ${REMOTE_BUILD_DIR}/{staging,ccache}"
+
+# Copy source
+echo "Uploading source..."
+rsync -az -e "ssh ${SSH_OPTS}" /home/node/node.tar.xz "${REMOTE_USER}@${REMOTE_HOST}:${REMOTE_BUILD_DIR}/staging/"
+
+# Create and execute build script remotely
+echo "Building on remote machine..."
+ssh ${SSH_OPTS} "${REMOTE_USER}@${REMOTE_HOST}" bash <<ENDSSH
```

Shellcheck (0.11.0)

[warning] Quote 'ENDSSH' to make here document expansions happen on the server side rather than on the client.

(SC2087)